packages/rum-core/src/domain/rumEventsCollection/view/trackViewMetrics.ts

```diff
@@ -8,6 +19,72 @@ import type { LifeCycle } from '../../lifeCycle'
 import { LifeCycleEventType } from '../../lifeCycle'
 import { waitPageActivityEnd } from '../../waitPageActivityEnd'
 
+import { getScrollY } from '../../../browser/scroll'
+import { getViewportDimension } from '../../../browser/viewportObservable'
+
+export interface ScrollMetrics {
+  maxScrollDepth: number
+  maxScrollHeight: number
+  maxScrollDepthTime: Duration
+  maxScrollTop: number
+}
+
+export const THROTTLE_SCROLL_DURATION = ONE_SECOND
+
+function computeScrollMetrics() {
+  const scrollTop = getScrollY()
+
+  const { height } = getViewportDimension()
+
+  const scrollHeight = Math.round((document.scrollingElement || document.documentElement).scrollHeight)
+  const scrollDepth = Math.round(height + scrollTop)
+
+  return {
+    scrollHeight,
+    scrollDepth,
+    scrollTop,
+  }
+}
+
+export function trackScrollMetrics(
+  viewStart: ClocksState,
+  callback: (scrollMetrics: ScrollMetrics) => void,
+  getMetrics = computeScrollMetrics
+) {
+  if (!isExperimentalFeatureEnabled(ExperimentalFeature.SCROLLMAP)) {
+    return { stop: noop }
+  }
+  let maxScrollDepth = 0
+  const handleScrollEvent = throttle(
+    () => {
+      const { scrollHeight, scrollDepth, scrollTop } = getMetrics()
+
+      if (scrollDepth > maxScrollDepth) {
+        const now = relativeNow()
+        const timeStamp = elapsed(viewStart.relative, now)
+        maxScrollDepth = scrollDepth
+        callback({
+          maxScrollDepth,
+          maxScrollHeight: scrollHeight,
+          maxScrollDepthTime: timeStamp,
+          maxScrollTop: scrollTop,
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**❗ warning:** ‏can't `scrollTop` and `scrollHeight` decrease?

Reply:
Changed the scroll metrics attributes naming: https://docs.google.com/document/d/16HYdn0JBH19xPEZRmnfTJfCYyE2LzwERzUlBTG7EcDs/edit